### PR TITLE
WIP: Add precompiles

### PR DESCRIPTION
**Pull request by timholy**
I'm not certain how much this helps (it wasn't very visible), but FWIW.

### CODE REVIEW

**Comment by timholy** [comment]
Incidentally, SnoopCompile generated ~500 precompiles for Base, which I could submit to julia proper. But for those too, it didn't seem to make a blatantly-obvious difference (It's a little hard to time things with Gallium, since it's interactive usage). So I held back from submitting those.

The one that seemed to make a pretty big difference was JuliaParser.

File: src/precompile.jl [hunk 1]
```diff
@@ -0,0 +1,61 @@
+function _precompile_()
```
Comment (Keno) [inline src/precompile.jl]: Don't you need to call this function somewhere?

Comment (timholy) [inline src/precompile.jl]: It's possible that could help just a little.

**Comment by timholy** [comment]
Interestingly, if I add a call to `_precompile_` in ELF I get a segfault, and adding one in DWARF (without doing it in ELF) yields
```
err = ErrorException("ELF.SectionRef must implement deref")

 [inlined code] from ./error.jl:22
 in _searchforip(::DWARF.UnitIterator{DWARF.DWARFCUHeader}, ::UInt64) at /home/tim/.julia/v0.5/DWARF/src/navigate.jl:16
 in searchcuforip(::ObjFileBase.DebugSections{ELF.ELFHandle{Base.AbstractIOBuffer{Array{UInt8,1}}},ELF.SectionRef}, ::UInt64) at /home/tim/.julia/v0.5/DWARF/src/navigate.jl:25
 in (::Gallium.##5#7)(::Array{UInt8,1}) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:183
 in rec_backtrace(::Gallium.##5#7, ::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Hooking/backtraces.jl:54
 in stackwalk(::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:148
 in breakpoint_hit(::Gallium.Hooking.Hook, ::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:290
 in _callback(::Ptr{Void}) at /home/tim/.julia/v0.5/Gallium/src/Hooking/Hooking.jl:153
 in callback(::Ptr{Void}) at /home/tim/.julia/v0.5/Gallium/src/Hooking/Hooking.jl:184
 in sinthesin(::Int64) at /home/tim/.julia/v0.5/Gallium/examples/testprograms/misc.jl:0
 in eval(::Module, ::Any) at ./boot.jl:237err = ErrorException("ELF.SectionRef must implement deref")

 [inlined code] from ./error.jl:22
 in _searchforip(::DWARF.UnitIterator{DWARF.DWARFCUHeader}, ::UInt64) at /home/tim/.julia/v0.5/DWARF/src/navigate.jl:16
 in searchcuforip(::ObjFileBase.DebugSections{ELF.ELFHandle{Base.AbstractIOBuffer{Array{UInt8,1}}},ELF.SectionRef}, ::UInt64) at /home/tim/.julia/v0.5/DWARF/src/navigate.jl:25
 in (::Gallium.##5#7)(::Array{UInt8,1}) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:183
 in rec_backtrace(::Gallium.##5#7, ::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Hooking/backtraces.jl:57
 in stackwalk(::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:148
 in breakpoint_hit(::Gallium.Hooking.Hook, ::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:290
 in _callback(::Ptr{Void}) at /home/tim/.julia/v0.5/Gallium/src/Hooking/Hooking.jl:153
 in callback(::Ptr{Void}) at /home/tim/.julia/v0.5/Gallium/src/Hooking/Hooking.jl:184
 in sinthesin(::Int64) at /home/tim/.julia/v0.5/Gallium/examples/testprograms/misc.jl:0
 in eval(::Module, ::Any) at ./boot.jl:237err = ErrorException("ELF.SectionRef must implement deref")

 [inlined code] from ./error.jl:22
 in _searchforip(::DWARF.UnitIterator{DWARF.DWARFCUHeader}, ::UInt64) at /home/tim/.julia/v0.5/DWARF/src/navigate.jl:16
 in searchcuforip(::ObjFileBase.DebugSections{ELF.ELFHandle{Base.AbstractIOBuffer{Array{UInt8,1}}},ELF.SectionRef}, ::UInt64) at /home/tim/.julia/v0.5/DWARF/src/navigate.jl:25
 in (::Gallium.##5#7)(::Array{UInt8,1}) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:183
 in rec_backtrace(::Gallium.##5#7, ::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Hooking/backtraces.jl:59
 in stackwalk(::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:148
 in breakpoint_hit(::Gallium.Hooking.Hook, ::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:290
 in _callback(::Ptr{Void}) at /home/tim/.julia/v0.5/Gallium/src/Hooking/Hooking.jl:153
 in callback(::Ptr{Void}) at /home/tim/.julia/v0.5/Gallium/src/Hooking/Hooking.jl:184
 in sinthesin(::Int64) at /home/tim/.julia/v0.5/Gallium/examples/testprograms/misc.jl:0
 in eval(::Module, ::Any) at ./boot.jl:237err = ErrorException("ELF.SectionRef must implement deref")

 [inlined code] from ./error.jl:22
 in _searchforip(::DWARF.UnitIterator{DWARF.DWARFCUHeader}, ::UInt64) at /home/tim/.julia/v0.5/DWARF/src/navigate.jl:16
 in searchcuforip(::ObjFileBase.DebugSections{ELF.ELFHandle{Base.AbstractIOBuffer{Array{UInt8,1}}},ELF.SectionRef}, ::UInt64) at /home/tim/.julia/v0.5/DWARF/src/navigate.jl:25
 in (::Gallium.##5#7)(::Array{UInt8,1}) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:183
 in rec_backtrace(::Gallium.##5#7, ::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Hooking/backtraces.jl:59
 in stackwalk(::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:148
 in breakpoint_hit(::Gallium.Hooking.Hook, ::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:290
 in _callback(::Ptr{Void}) at /home/tim/.julia/v0.5/Gallium/src/Hooking/Hooking.jl:153
 in callback(::Ptr{Void}) at /home/tim/.julia/v0.5/Gallium/src/Hooking/Hooking.jl:184
 in sinthesin(::Int64) at /home/tim/.julia/v0.5/Gallium/examples/testprograms/misc.jl:0
 in eval(::Module, ::Any) at ./boot.jl:237err = ErrorException("ELF.SectionRef must implement deref")

 [inlined code] from ./error.jl:22
 in _searchforip(::DWARF.UnitIterator{DWARF.DWARFCUHeader}, ::UInt64) at /home/tim/.julia/v0.5/DWARF/src/navigate.jl:16
 in searchcuforip(::ObjFileBase.DebugSections{ELF.ELFHandle{Base.AbstractIOBuffer{Array{UInt8,1}}},ELF.SectionRef}, ::UInt64) at /home/tim/.julia/v0.5/DWARF/src/navigate.jl:25
 in (::Gallium.##5#7)(::Array{UInt8,1}) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:183
 in rec_backtrace(::Gallium.##5#7, ::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Hooking/backtraces.jl:59
 in stackwalk(::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:148
 in breakpoint_hit(::Gallium.Hooking.Hook, ::Gallium.Hooking.RegisterContext) at /home/tim/.julia/v0.5/Gallium/src/Gallium.jl:290
 in _callback(::Ptr{Void}) at /home/tim/.julia/v0.5/Gallium/src/Hooking/Hooking.jl:153
 in callback(::Ptr{Void}) at /home/tim/.julia/v0.5/Gallium/src/Hooking/Hooking.jl:184
 in sinthesin(::Int64) at /home/tim/.julia/v0.5/Gallium/examples/testprograms/misc.jl:0
 in eval(::Module, ::Any) at ./boot.jl:2371 @noinline function sinthesin(x)
```

I'm not going to be able to pursue this any more today, and perhaps for a week---I have a ton of teaching next week.

**Comment by Keno** [comment]
I added my own set of precompiles a while back, so I'll close this.

**Pull request closed without merging**